core/src/main/scala/chisel3/StrongEnum.scala

```diff
@@ -2,74 +2,18 @@
+import chisel3.internal.{Binding, Builder, ChildBinding, ConstrainedBinding, DynamicSelectedElementBinding, InstanceId, SampleElementBinding, throwException}
 import firrtl.annotations._
+import firrtl.transforms.{CustomRadixApplyAnnotation, CustomRadixDefAnnotation}
 
-
-object EnumAnnotations {
-  /** An annotation for strong enum instances that are ''not'' inside of Vecs
-    *
-    * @param target the enum instance being annotated
-    * @param enumTypeName the name of the enum's type (e.g. ''"mypackage.MyEnum"'')
-    */
-  case class EnumComponentAnnotation(target: Named, enumTypeName: String) extends SingleTargetAnnotation[Named] {
-    def duplicate(n: Named): EnumComponentAnnotation = this.copy(target = n)
-  }
-
-  case class EnumComponentChiselAnnotation(target: InstanceId, enumTypeName: String) extends ChiselAnnotation {
-    def toFirrtl: EnumComponentAnnotation = EnumComponentAnnotation(target.toNamed, enumTypeName)
-  }
-
-  /** An annotation for Vecs of strong enums.
-    *
-    * The ''fields'' parameter deserves special attention, since it may be difficult to understand. Suppose you create a the following Vec:
-
-    *               {{{
-    *               VecInit(new Bundle {
-    *                 val e = MyEnum()
-    *                 val b = new Bundle {
-    *                   val inner_e = MyEnum()
-    *                 }
-    *                 val v = Vec(3, MyEnum())
-    *               }
-    *               }}}
-    *
-    *               Then, the ''fields'' parameter will be: ''Seq(Seq("e"), Seq("b", "inner_e"), Seq("v"))''. Note that for any Vec that doesn't contain Bundles, this field will simply be an empty Seq.
-    *
-    * @param target the Vec being annotated
-    * @param typeName the name of the enum's type (e.g. ''"mypackage.MyEnum"'')
-    * @param fields a list of all chains of elements leading from the Vec instance to its inner enum fields.
-    *
-    */
-  case class EnumVecAnnotation(target: Named, typeName: String, fields: Seq[Seq[String]]) extends SingleTargetAnnotation[Named] {
-    def duplicate(n: Named): EnumVecAnnotation = this.copy(target = n)
-  }
-
-  case class EnumVecChiselAnnotation(target: InstanceId, typeName: String, fields: Seq[Seq[String]]) extends ChiselAnnotation {
-    override def toFirrtl: EnumVecAnnotation = EnumVecAnnotation(target.toNamed, typeName, fields)
-  }
-
-  /** An annotation for enum types (rather than enum ''instances'').
-    *
-    * @param typeName the name of the enum's type (e.g. ''"mypackage.MyEnum"'')
-    * @param definition a map describing which integer values correspond to which enum names
-    */
-  case class EnumDefAnnotation(typeName: String, definition: Map[String, BigInt]) extends NoTargetAnnotation
-
-  case class EnumDefChiselAnnotation(typeName: String, definition: Map[String, BigInt]) extends ChiselAnnotation {
-    override def toFirrtl: Annotation = EnumDefAnnotation(typeName, definition)
-  }
-}
-import EnumAnnotations._
-
```

Comment:
I'm down with the basic idea, but why delete these? I think they should still exist but be deprecated. We should also consider having a command-line option to continue generating them (at least for 3.5) but we can maybe just see if anyone complains before adding that.

Reply:
Should I'll add them back
deprecate them form Chisel 3.5
add a command-line option to emit them(but by default turn it off)
then remove them in Chisel 3.6.

Reply:
@jackkoenig In #2240, I deprecate them from 3.4.x, and remove it in 3.5. Is this OK? or we should deprecate it since 3.5 and remove them in 3.6?

Reply:
Because it's so late in 3.5 release process (deleting anything else is another risk that might push back `3.5.0` release), I think we should deprecate in 3.5 and delete in 3.6.

Let's also make the 3.5->3.6 time much faster, 6 months or less this time!